### ISSUE TITLE
query storage value - height off by 1 from what's expected in tendermint

### ISSUE THREAD

**Issue by tzemanovic** [anoma/namada#709]
According to tendermint's abci_query expectation, we should be reading state from `height - 1` when not `0` (latest committed height).

https://github.com/tendermint/tendermint/blob/v0.34.x/spec/abci/abci.md#query-1 : “Note that this is the height of the block containing the application’s Merkle root hash, which represents the state as it was after committing the block at Height-1”

**Comment by cwgoes** [comment]
Does this apply to all Namada state queries at the moment?

**Comment by tzemanovic** [comment]
> What does this apply to in the Namada codebase? Places where we're using `abci_query` in the client?

Yeah, any `abci_query` that supports explicit height (some are pending on #172). I'm a bit confused by this as it seems a bit counter intuitive, but not sure if this is a hard requirement or a recommendation
